### PR TITLE
DateTime: Fix empty dates

### LINKED ISSUE #6912: Date field with empty value shows date in frontend
### Description

I created a Date field and enabled the Allow empty value option, also the HTML5 input field, as I read somewhere that would help. Unfortunately if the field is empty it shows a strange date now in the frontend.
How to avoid that?
Also the date is shown in English tho my Wordpress is in German. Is there a way to translate that?

### Version

2.9.3

### Testing Instructions

_No response_

### Screenshots / Screencast

Screenshot 2022-09-20 at 09 43 20
Screenshot 2022-09-20 at 09 38 55

### Possible Workaround

_No response_

### Site Health Information

```text
`
### wp-core ###

version: 6.0.2
site_language: de_CH_informal
user_language: de_CH_informal
timezone: Europe/Zurich
permalink: /%postname%/
https_status: true
multisite: false
user_registration: 0
blog_public: 0
default_comment_status: open
environment_type: production
user_count: 2
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /home/ncbich/www/www.ncbi.ch/wp2022
wordpress_size: loading...
uploads_path: /home/ncbich/www/www.ncbi.ch/wp2022/wp-content/uploads
uploads_size: loading...
themes_path: /home/ncbich/www/www.ncbi.ch/wp2022/wp-content/themes
themes_size: loading...
plugins_path: /home/ncbich/www/www.ncbi.ch/wp2022/wp-content/plugins
plugins_size: loading...
database_size: loading...
total_size: loading...

### wp-active-theme ###

name: Hello Elementor (hello-elementor)
version: 2.6.1
author: Elementor Team
author_website: https://elementor.com/?utm_source=wp-themes&utm_campaign=author-uri&utm_medium=wp-dash
parent_theme: none
theme_features: core-block-patterns, widgets-block-editor, post-thumbnails, menus, automatic-feed-links, title-tag, html5, custom-logo, editor-style, align-wide, woocommerce, wc-product-gallery-zoom, wc-product-gallery-lightbox, wc-product-gallery-slider
theme_path: /home/ncbich/www/www.ncbi.ch/wp2022/wp-content/themes/hello-elementor
auto_update: Deaktiviert

### wp-themes-inactive (1) ###

Twenty Twenty-Two: version: 1.2, author: WordPress-Team, Automatische Aktualisierungen deaktiviert

### wp-plugins-active (17) ###

Advanced Post Queries: version: 1.0.16, author: Shabti Kaplan, Automatische Aktualisierungen deaktiviert
Autoptimize: version: 3.1.2, author: Frank Goossens (futtta), Automatische Aktualisierungen aktiviert
BackWPup: version: 3.10.0, author: Inpsyde GmbH, Automatische Aktualisierungen aktiviert
Classic Editor: version: 1.6.2, author: WordPress Contributors, Automatische Aktualisierungen aktiviert
Dynamic Visibility for Elementor: version: 5.0.1, author: Dynamic.ooo, Automatische Aktualisierungen aktiviert
Ele Custom Skin: version: 3.1.7, author: Dudaster.com, Automatische Aktualisierungen deaktiviert
Elementor: version: 3.8.0-dev4, author: Elementor.com, Automatische Aktualisierungen aktiviert
Elementor Beta (Developer Edition): version: 1.1.1, author: Elementor.com, Automatische Aktualisierungen aktiviert
Elementor Pro: version: 3.8.0-dev42, author: Elementor.com, Automatische Aktualisierungen aktiviert
FileBird: version: 5.0.7, author: Ninja Team, Automatische Aktualisierungen aktiviert
Huger for Elementor: version: 1.1.1, author: Merkulove, Automatische Aktualisierungen aktiviert
Pods - Custom Content Types and Fields: version: 2.9.3, author: Pods Framework Team, Automatische Aktualisierungen aktiviert
Rank Math SEO: version: 1.0.97, author: Rank Math, Automatische Aktualisierungen aktiviert
Table Rate Shipping for WooCommerce: version: 2.26, author: JEM Plugins, Automatische Aktualisierungen aktiviert
WooCommerce: version: 6.9.2, author: Automattic, Automatische Aktualisierungen aktiviert
Wordfence Security: version: 7.6.2, author: Wordfence, Automatische Aktualisierungen aktiviert
zahls.ch Credit Cards, PostFinance and TWINT for WooCommerce: version: 1.2.0, author: siebenberge gmbh, Automatische Aktualisierungen aktiviert

### wp-plugins-inactive (1) ###

Jetpack: version: 11.3.1, author: Automattic, Automatische Aktualisierungen aktiviert

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1808
imagemagick_version: ImageMagick 7.1.0-34 Q16-HDRI amd64 20058 https://imagemagick.org
imagick_version: 3.5.1
file_uploads: File uploads is turned off
post_max_size: 128M
upload_max_filesize: 64M
max_effective_size: 64 MB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 384 GB
	imagick::RESOURCETYPE_DISK: 9.2233720368548E+18
	imagick::RESOURCETYPE_FILE: 1536
	imagick::RESOURCETYPE_MAP: 384 GB
	imagick::RESOURCETYPE_MEMORY: 192 GB
	imagick::RESOURCETYPE_THREAD: 1
imagemagick_file_formats: 3FR, 3G2, 3GP, AAI, AI, APNG, ART, ARW, ASHLAR, AVI, AVIF, AVS, BGR, BGRA, BGRO, BIE, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CR3, CRW, CUBE, CUR, CUT, DATA, DCM, DCR, DCRAW, DCX, DDS, DFONT, DNG, DOT, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, EXR, FARBFELD, FAX, FF, FILE, FITS, FL32, FLV, FRACTAL, FTP, FTS, FTXT, G3, G4, GIF, GIF87, GRADIENT, GRAY, GRAYA, GROUP4, GV, HALD, HDR, HEIC, HEIF, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, JXL, K25, KDC, KERNEL, LABEL, M2V, M4V, MAC, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MSVG, MTV, MVG, NEF, NRW, NULL, ORA, ORF, OTB, OTF, PAL, PALM, PAM, PANGO, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PGX, PHM, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, POCKETMOD, PPM, PS, PS2, PS3, PSB, PSD, PTIF, PWP, QOI, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGB565, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, STRIMG, SUN, SVG, SVGZ, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TM2, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WEBM, WEBP, WMF, WMV, WMZ, WPG, X3F, XBM, XC, XCF, XPM, XPS, XV, YAML, YCbCr, YCbCrA, YUV
gd_version: 2.3.1
gd_formats: GIF, JPEG, PNG, WebP, BMP
ghostscript_version: 9.56.1

### wp-server ###

server_architecture: FreeBSD 12.3-RELEASE-p6 amd64
httpd_software: Apache
php_version: 7.4.30 64bit
php_sapi: cgi-fcgi
max_input_variables: 2000
time_limit: 30
memory_limit: 512M
max_input_time: 60
upload_max_filesize: 64M
php_post_max_size: 128M
curl_version: 7.85.0 OpenSSL/1.1.1q
suhosin: false
imagick_availability: true
pretty_permalinks: true
htaccess_extra_rules: true

### wp-database ###

extension: mysqli
server_version: 10.3.36-MariaDB-log
client_version: mysqlnd 7.4.30
max_allowed_packet: 16777216
max_connections: 400

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /home/ncbich/www/www.ncbi.ch/wp2022/wp-content
WP_PLUGIN_DIR: /home/ncbich/www/www.ncbi.ch/wp2022/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 512M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: Nicht definiert
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

### pods ###

pods-server-software: Apache
pods-user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:104.0) Gecko/20100101 Firefox/104.0
pods-session-save-path: /var/cache/php-sessions
pods-session-save-path-exists: Yes
pods-session-save-path-writable: Yes
pods-session-max-lifetime: 1440
pods-opcode-cache-apc: Nein
pods-opcode-cache-memcached: Nein
pods-opcode-cache-opcache: Yes
pods-opcode-cache-redis: Nein
pods-object-cache-apc: Nein
pods-object-cache-apcu: Yes
pods-object-cache-memcache: Nein
pods-object-cache-memcached: Nein
pods-object-cache-redis: Nein
pods-memory-current-usage: 117.869M
pods-memory-current-usage-real: 102.000M
pods-network-wide: Nein
pods-install-location: /wp-content/plugins/pods/
pods-developer: Nein
pods-tableless-mode: Nein
pods-relationship-table-enabled: Yes
pods-light-mode: Nein
pods-strict: Nein
pods-allow-deprecated: Yes
pods-api-cache: Yes
pods-shortcode-allow-evaluate-tags: Nein
pods-sessions: Disable sessions
pods-can-use-sessions: Yes
pods-relationship-table-status: 2

`
```

### Pods Package

```json
{
    "@meta": {
        "version": "2.9.3",
        "build": 1663660262
    },
    "settings": {
        "types_only": "0",
        "watch_changed_fields": "0",
        "metadata_integration": "1",
        "metadata_override_get": "0",
        "session_auto_start": "0",
        "wisdom_opt_out": ""
    },
    "pods": [
        {
            "name": "veranstaltung",
            "id": 1854,
            "label": "Veranstaltungen",
            "description": "",
            "type": "post_type",
            "storage": "meta",
            "label_singular": "Veranstaltung",
            "public": "1",
            "show_ui": "1",
            "rest_enable": "1",
            "supports_title": "1",
            "supports_editor": "1",
            "_migrated_28": "1",
            "show_in_menu": "1",
            "menu_position": "0",
            "show_in_nav_menus": "1",
            "show_in_admin_bar": "1",
            "menu_icon": "dashicons-calendar-alt",
            "has_archive": "1",
            "required": "0",
            "repeatable": "0",
            "publicly_queryable": "1",
            "groups": [
                {
                    "name": "more_fields",
                    "id": 1855,
                    "label": "More Fields",
                    "description": "",
                    "weight": 0,
                    "fields": [
                        {
                            "name": "untertitel",
                            "id": 1869,
                            "label": "Untertitel",
                            "description": "",
                            "weight": 0,
                            "type": "text",
                            "sister_id": "-- Select One --",
                            "required": "0",
                            "repeatable": "0",
                            "text_allowed_html_tags": "strong em a ul ol li b i",
                            "text_max_length": "255",
                            "roles_allowed": "administrator",
                            "rest_pick_response": "array",
                            "rest_pick_depth": "1"
                        },
                        {
                            "name": "datum",
                            "id": 1872,
                            "label": "Datum",
                            "description": "",
                            "weight": 1,
                            "type": "date",
                            "sister_id": "-- Select One --",
                            "required": "1",
                            "repeatable": "0",
                            "roles_allowed": "administrator",
                            "rest_pick_response": "array",
                            "rest_pick_depth": "1",
                            "date_type": "wp",
                            "date_format": "mdy",
                            "date_allow_empty": "1",
                            "date_html5": "0"
                        },
                        {
                            "name": "enddatum",
                            "id": 1873,
                            "label": "Enddatum",
                            "description": "",
                            "weight": 2,
                            "type": "date",
                            "sister_id": "-- Select One --",
                            "required": "0",
                            "repeatable": "0",
                            "roles_allowed": "administrator",
                            "rest_pick_response": "array",
                            "rest_pick_depth": "1",
                            "date_type": "wp",
                            "date_format": "mdy",
                            "date_allow_empty": "1",
                            "date_html5": "1"
                        },
                        {
                            "name": "ort",
                            "id": 1874,
                            "label": "Ort",
                            "description": "",
                            "weight": 3,
                            "type": "text",
                            "sister_id": "-- Select One --",
                            "required": "1",
                            "repeatable": "0",
                            "text_allowed_html_tags": "strong em a ul ol li b i",
                            "text_max_length": "255",
                            "roles_allowed": "administrator",
                            "rest_pick_response": "array",
                            "rest_pick_depth": "1"
                        },
                        {
                            "name": "ausschreibung",
                            "id": 1858,
                            "label": "Ausschreibung",
                            "description": "",
                            "weight": 4,
                            "type": "file",
                            "sister_id": "-- Select One --",
                            "required": "0",
                            "repeatable": "0",
                            "file_format_type": "single",
                            "file_uploader": "attachment",
                            "file_type": "other",
                            "file_attachment_tab": "upload",
                            "file_upload_dir": "wp",
                            "file_edit_title": "1",
                            "file_show_edit_link": "0",
                            "file_linked": "0",
                            "file_limit": "0",
                            "file_field_template": "rows",
                            "file_add_button": "Add File",
                            "file_modal_title": "Attach a file",
                            "file_modal_add_button": "Add File",
                            "file_wp_gallery_link": "file",
                            "file_wp_gallery_columns": "3",
                            "file_wp_gallery_size": "thumbnail",
                            "roles_allowed": "administrator",
                            "rest_pick_response": "array",
                            "rest_pick_depth": "1",
                            "file_allowed_extensions": "pdf"
                        },
                        {
                            "name": "anmeldung",
                            "id": 1859,
                            "label": "Anmeldung",
                            "description": "",
                            "weight": 5,
                            "type": "text",
                            "sister_id": "-- Select One --",
                            "required": "0",
                            "repeatable": "0",
                            "text_allowed_html_tags": "strong em a ul ol li b i",
                            "text_max_length": "255",
                            "roles_allowed": "administrator",
                            "rest_pick_response": "array",
                            "rest_pick_depth": "1",
                            "text_allow_shortcode": "1"
                        }
                    ]
                }
            ]
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by JoryHogeveen**
## Description

This PR makes sure React allows empty dates if the user allowed it in the field settings.

## Related GitHub issue(s)

Fixes #6912 

## PR checklist

- [ ] I have tested my own code to confirm it works as I intended.
- [ ] My code follows the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [ ] My code follows the [WordPress Inline Documentation Standards](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/).
- [ ] My code includes automated tests for PHP and/or JS (if applicable).

### CODE REVIEW

**Comment by JoryHogeveen** [comment]
@sc0ttkclark
Please review this PR thoroughly since it changes the way we store empty dates.
Currently we always store a date, even if empty (0000-00-00 etc). With this PR this changes. We still allow storage of such empty dates but by default it will simply store an empty string.
This will improve compatibility with third party tools for empty checks while AFAIK it will also be backwards compatible since we check for an empty string as well in `PodsField_DateTime::is_empty()`

File: ui/js/dfv/src/fields/datetime/index.js [hunk 1]
```diff
@@ -119,6 +119,9 @@ const DateTime = ( {
 
 	const useHTML5Field = toBool( html5 ) && checkForHTML5BrowserSupport( 'datetime-local' );
 
+	const allowEmpty = toBool( fieldConfig.datetime_allow_empty ?? 0 );
+	const emptyValues = [ '0000-00-00', '0000-00-00 00:00:00', '00:00:00', '' ];
```
Comment (sc0ttkclark) [inline ui/js/dfv/src/fields/datetime/index.js:123]: The exception here is that 00:00:00 is a valid non-empty time right?

Comment (JoryHogeveen) [reply ui/js/dfv/src/fields/datetime/index.js:123]: Check, I can also move the valid empty values to the appropriate classes and let the `date` and `time` classes override the `datetime` main class.

**Review by sc0ttkclark — APPROVED** [review]

File: ui/js/dfv/src/fields/datetime/index.js (hunk 1 above)
Comment (sc0ttkclark) [reply ui/js/dfv/src/fields/datetime/index.js]: 00:00:00 is an intentionally supported value and is treated as non-empty in other areas. I've removed it from the list here.

**Review by sc0ttkclark — DISMISSED** [review]

**Pull request closed without merging**